sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveReferencesInAggregate.scala

```diff
@@ -112,9 +105,11 @@ object ResolveReferencesInAggregate extends SQLConfHelper
     assert(selectList.forall(_.resolved))
     if (conf.groupByAliases) {
       groupExprs.map { g =>
-        g.transformWithPruning(_.containsPattern(UNRESOLVED_ATTRIBUTE)) {
-          case u: UnresolvedAttribute =>
-            selectList.find(ne => conf.resolver(ne.name, u.name)).getOrElse(u)
+        g.transformWithPruning(_.containsAnyPattern(UNRESOLVED_ATTRIBUTE,
+          LATERAL_COLUMN_ALIAS_REFERENCE)) {
+          case u @ (_: UnresolvedAttribute | _: LateralColumnAliasReference) =>
```

Comment:
shall we update the lca resolution rule to not resolve columns in grouping expressions to `LateralColumnAliasReference`? The produced plan is very weird. also cc  @anchovYu

Reply:
Thanks @cloud-fan , I'll try to implement it in followup PR, please help to see if it meets the requirements when it ready.